### PR TITLE
add bt xml config and plugins

### PR DESCRIPTION
**Pull request by Marwan99**
per #21

### CODE REVIEW

**Comment by Marwan99** [comment]
@SteveMacenski Just want an okay from you to go ahead with this format. A more or less ready example to check is the Backup action.

**Comment by SteveMacenski** [comment]
yeah this example looks great. I think you just need a couple sentences at the top of each page for what each of them does, and a bunch are missing examples. Otherwise, LGTM.

File: configuration/packages/configuring-bt-xml.rst [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.. _configuring_behavior_tree:
+.. _configuring_behavior_tree_xml:
 
 Configuring the behavior tree XML
```
Comment (SteveMacenski) [inline configuration/packages/configuring-bt-xml.rst:3]:
```suggestion
Behavior Tree XML Nodes
```

Comment (SteveMacenski) [reply configuration/packages/configuring-bt-xml.rst:3]: This title is a little long and not in the same theme as the others

**Comment by Marwan99** [comment]
Still a couple more examples and descriptions need to be added. Will call it a day today and finish this tomorrow (technically it's already tomorrow my time xD).

**Comment by SteveMacenski** [comment]
OK. Only thing I notice is that there's no parameters listed (e.g. non input ports, but actual ROS parameters that individual nodes use in the BT navigator node namespace). Listing the parameters in each BT node separately of port would be nice since a few have them I think.

For the cases where the input port is taken from the blackboard, I don't see that represented in anyway. I think `goal` and `path` are the only 2 though in compute path to pose so that's a small update

**Comment by Marwan99** [comment]
I had a look at the param list and the only params I think are relevant here are the recoveries but I can see you adding them in #26. Can you please point me to the ones you are talking about?

**Comment by SteveMacenski** [comment]
I did a run through of all of the BT nodes, these 2 have parameters from the BT nav

https://github.com/ros-planning/navigation2/blob/master/nav2_behavior_tree/plugins/condition/goal_reached_condition.cpp

https://github.com/ros-planning/navigation2/blob/master/nav2_behavior_tree/plugins/condition/distance_traveled_condition.cpp

I wonder if we should at minimum make a note that they use the BT navigator's value for these parameters, so they know it effects it

**Comment by Marwan99** [comment]
A note as you mentioned with maybe a link to the bt navigator params might do it. I will also need to add a page for the distance travelled condition.

**Comment by SteveMacenski** [comment]
Oh I didn't check if you included them all,  I assumed you had

**Comment by Marwan99** [comment]
Well.. I was assuming everything was in the param file, sorry about that. Will go through the code and make sure we have everything.

**Comment by SteveMacenski** [comment]
Did you add the 2 nodes with parameters yet? I think after that this is good to ship now that you added the missing nodes

**Comment by Marwan99** [comment]
More things coming, give me a couple of minutes.

**Comment by Marwan99** [comment]
Do we need to have `name` added as input port? 

Other than I think it is ready for review. Some of the conditions and the round robin action need descriptions and/or descriptions for their parameter, as those were missed in the param md file and didn't find documentation for them.

**Comment by SteveMacenski** [comment]
What do you mean by `name`?

**Comment by Marwan99** [comment]
Sorry `name` is just an attribute not a port, but it is optional https://www.behaviortree.dev/xml_format/.

Examples of `name` usage in a bt:
https://github.com/ros-planning/navigation2/blob/caea7fa4e68be18f6018796b983c8d008a8b492e/nav2_bt_navigator/behavior_trees/navigate_w_replanning_and_recovery.xml#L8-L13

**Comment by SteveMacenski** [comment]
Ah ok. I think its fine to not include it, but add a note on the parent page of the BT node sub-pages that they all have this `name` attribute for identifying in logging. 

Which of the files did you need descriptions / examples for? If you enumerate them, I'll provide them.

**Comment by Marwan99** [comment]
- Wait action (I think it needs improving)
- ReinitializeGlobalLocalization action (missing description)
- DistanceTraveled condition (missing description and distance param description)
- InitialPoseReceived condition (missing description)
- TimeExpired condition (missing description and seconds param description)
- RoundRobin control (missing description)

**Comment by SteveMacenski** [comment]
Reinitialize description: Used to trigger global relocalization using AMCL in case of severe delocalization or kidnapped robot problem.

Distance Traveled: Node that returns success when a configurable distance has been traveled.
(the distance parameter is the distance it must travel before returning success (m) )

Initial Pose Received: Node that returns success when the initial pose is sent to AMCL via `/initial_pose`.

Time Expired: Node that returns success when a time duration has passed
(seconds is the time passed to return success (s) )
RoundRobin: Custom control flow node used to create a round-robbin behavior for children BT nodes.

**Comment by SteveMacenski** [comment]
Let me know when I should look again

**Comment by Marwan99** [comment]
I think its ready for reviewing.

**Comment by SteveMacenski** [comment]
Can you send me a link to the index.html for this? I can't seem to find it

**Comment by Marwan99** [comment]
https://164-240847415-gh.circle-artifacts.com/0/html/index.html

**Pull request merged**